### ISSUE TITLE
sub-modules of `exegol-resources` are missing

### ISSUE THREAD

**Issue by LucasVanHaaren** [ThePorgs/Exegol#180]
### Describe the bug

Hi folks,

I just did a fresh install of exegol and find that all exegol-resources which are sub-modules are missing in my `~/.exegol/exegol-resources`.
I have been able to reproduce in another environment using an older exegol version.

I'm currently looking at the codebase to identify where the problem might be coming from.

Thanks in advance :love_you_gesture: 

### Steps To Reproduce

1. Install exegol
2. Install an image
3. List files on host in `~/.exegol/exegol-resources/webshells/PHP/p0wny-shell/` => empty
4. List files on host in `~/.exegol/exegol-resources/encrypted_disks/bitleaker/` => empty
5. List files on host in `~/.exegol/exegol-resources/linux/mimipy/` => empty

### Exegol Wrapper Version

```Text
[*] Exegol is currently in version v4.2.5
[*] Exegol Discord serv.: https://discord.gg/cXThyp7D6P
[*] Exegol documentation: https://exegol.rtfd.io/
[+] We thank Capgemini for supporting the project (helping with dev) 🙏                                                                                                                       
[+] We thank HackTheBox for sponsoring the multi-arch support 💚                                                                                                                              
[D] Pip installation: On ✔                                                                                                                                                                    
[D] Git source installation: Off 🪓                                                                                                                                                           
[D] Host OS: HostOs.LINUX (DockerEngine.LINUX)                                                                                                                                                
[D] Arch: amd64                                                                                                                                                                               
[D] Raw arch: x86_64                                                                                                                                                                          
[D] Docker desktop: Off 🪓                                                                                                                                                                    
[D] Shell type: HostOs.LINUX                                                                                                                                                                  
[D] Last wrapper update check: 2023-09-16 00:00:00
```

### Host OS

Linux

### Configuration of the concerned container

```Text
┌──────────────────┬───────────────────────────────────────────────────────────────────┐
│             Name │ test                                                              │
│            Image │ full - v.3.1.1 (Up to date) (amd64)                               │
├──────────────────┼───────────────────────────────────────────────────────────────────┤
│    Creation date │ 16/09/2023 20:43                                                  │
│              GUI │ On ✔                                                              │
│          Network │ host                                                              │
│         Timezone │ On ✔                                                              │
│ Exegol resources │ On ✔ (/opt/resources)                                             │
│     My resources │ On ✔ (/opt/my-resources)                                          │
│    Shell logging │ Off 🪓                                                            │
│       Privileged │ Off ✔                                                             │
│        Workspace │ Dedicated (/workspace)                                            │
│             Envs │ DISPLAY=:0                                                        │
│                  │ _JAVA_AWT_WM_NONREPARENTING=1                                     │
│                  │ QT_X11_NO_MITSHM=1                                                │
│                  │ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin │
│          Volumes │ /tmp/.X11-unix ➡ /tmp/.X11-unix                                   │
│                  │ /etc/timezone ➡ /etc/timezone (RO)                                │
│                  │ /etc/localtime ➡ /etc/localtime (RO)                              │
│                  │ /home/vhash/.exegol/my-resources ➡ /opt/my-resources              │
│                  │ /home/vhash/.exegol/exegol-resources ➡ /opt/resources             │
│                  │ /home/vhash/.exegol/workspaces/test ➡ /workspace                  │
└──────────────────┴───────────────────────────────────────────────────────────────────┘
```

### Execution logs in debug mode

```Text
No logs as the issue is visible without running a container
```

### Anything else?

![image](https://github.com/ThePorgs/Exegol/assets/29121316/e262b18e-819b-4d1b-84c1-27b64ef1eb05)

**Comment by LucasVanHaaren** [comment]
Hi, I managed to reproduce this bug in 2 other environments (on OSX, and Ubuntu 22), all on pip installations. 
Source installations don't seem to be affected, as all exegol-resources sub-modules are present during my testings !

I tried to look at the code in order to produce a fix, but I must admit I had some trouble understanding how the git wrapper was designed but I think I've identified the piece of code that do the update. Is it the `GitUtils.submoduleSourceUpdate` function that is used to update, even on pip installations ? 

If so, according to this documentation, the recursive update of sub-modules may be done manually : https://stackoverflow.com/questions/22949068/using-gitpython-how-do-i-do-git-submodule-update-init

But this doesn't explain why this appears only on pip installs ...

Apart from the resource path specified in the exegol configuration, is the initialization and updating of exegol-resources differ depending on the installation mode ?

**Comment by ShutdownRepo** [comment]
@Dramelac I'll let you handle this issue, you'll probably have more insight on this than me

**Comment by Dramelac** [comment]
> Apart from the resource path specified in the exegol configuration, is the initialization and updating of exegol-resources differ depending on the installation mode ?

Hello

Yes there is some difference, when using a source install, the ressources are a submoudle itself of the wrapper but when using a pip install, ressources are first git cloned and then submodules are supposed to be updated. 

I will take a look at this, thank you for raising this issue

**Comment by Dramelac** [comment]
Fixed in 9a54364e1d7e57fbe5e0b9ce71b715bc5e20e12e , will be available in the next Exegol release.
After the wrapper update, in order to force the update of `exegol-resources`, you just have to remove the folder, the wrapper will ask you to install `exegol-resources` again and everything should be good after that.

**Comment by LucasVanHaaren** [comment]
Nice job, thank you !
